salesforce-composite/2.7/modules/ROOT/pages/edits/salesforce-composite-connector-examples.adoc

```diff
@@ -0,0 +1,469 @@
+	</salesforce-composite:composite-config>
+	<flow name="salesforce-composite-create-sobject-trees">
+		<http:listener doc:name="Listener"
+      path="/createSObjectTree"
+      config-ref="HTTP_Listener_config"/>
+		<ee:transform doc:name="Transform HTTP input for Create SObject Tree">
+			<ee:message>
+				<ee:set-payload><![CDATA[%dw 2.0
+output application/java
+---
+payload.records map ((record , indexOfRecord) -> {
+	"attributes": {
+		"type": record.attributes.objectType,
+		"referenceId": record.attributes.referenceId
+	},
+	"Name": record.Name,
+	"Phone": record.Phone,
+	"Website": record.Website,
+	"NumberOfEmployees": record.NumberOfEmployees as Number,
+	("ChildAccounts": {
+		"records": record.ChildAccounts.records map ((record01, indexOfRecord01) -> {
+			"attributes": {
+				"type": record01.attributes.objectType,
+				"referenceId": record01.attributes.referenceId
+			},
+			"Name": record01.Name,
+			"Phone": record01.Phone,
+			"Website": record01.Website,
+			"NumberOfEmployees": record01.NumberOfEmployees as Number
+		})
+	}),
+	"Contacts": {
+		"records": record.Contacts.records map ((record01, indexOfRecord01) -> {
+			"attributes": {
+				"type": record01.attributes.objectType,
+				"referenceId": record01.attributes.referenceId
+			},
+			"LastName": record01.LastName,
+			"Email": record01.Email,
+			"Title": record01.Title
+		})
+	}
+})
+]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+		<salesforce-composite:create-sobject-tree
+      doc:name="Create SObject tree"
+      config-ref="Salesforce_Composite_Salesforce_Composite"
+      type="Account"/>
+		<ee:transform doc:name="Transform Message">
+			<ee:message >
+				<ee:set-payload ><![CDATA[%dw 2.0
+output application/json
+---
+payload]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+	</flow>
+</mule>
+----
+
+== Pre-Query Example
+
+This example provides metadata for an object to query, generates two subrequests that contain this metadata, and then executes the subrequests:
+
+* HTTP *Listener*
++
+Accepts data from HTTP requests
+* *Transform Message*
++
+Specifies name for object
+* *Pre create*
++
+Provides metadata for the object that is going to be created and generates a subrequest used in execution
+* *Transform Message*
++
+Specifies values for fields
+* *Pre query*
++
+Provides metadata for the object that is going to be queried and generates a subrequest used in execution
+* *Transform Message*
++
+Passes output from *Pre query* operation
+* *Execute composite batch*
++
+Sends all subrequests in one batch
+* *Transform Message*
++
+Outputs the results in JSON format
++
+.The flow for this example includes an HTTP Listener component, Transform components, and multiple Salesfore Composite operations.
+image::salesforce-composite-prequery-flow.png[Pre-Query flow in Studio]
+
+=== Add an HTTP Listener
+
+The HTTP *Listener* source operation initiates the flow when it detects an event in the `executeFlow` path:
```

Comment:
This tells what the operation does and ends with a colon indicating that the following steps tell what the operation does, not what the audience does. Reword for clarity.